### PR TITLE
chore: add information about Apple Silicon machines

### PR DESCRIPTION
**Pull request by jawnsy**
Added some documentation about Rosetta to emulate x86-64:

image

### CODE REVIEW

File: setup/docker.md [hunk 1]
```diff
@@ -17,7 +17,19 @@ Coder for Docker works with the following platforms:
   [Docker Desktop 20.10](https://www.docker.com/products/docker-desktop).
   **Note**: Coder for Docker requires Windows Subsystem for Linux at this time.
 
-Coder requires x86-64 and does not support ARM-based processors at this time.
+At this time, Coder publishes builds for the x86-64 architecture only, and does
+not support support Arm-based processors, such as those present in Apple Silicon
+(M1) laptops or Amazon Graviton instances.
+
+On Apple Silicon laptops, we have had some success using [Rosetta 2] to emulate
```
Comment (jawnsy) [inline setup/docker.md:24]: We may want to move this into a bullet point note as we do for Windows 11?

**Review by johnstcn — APPROVED** [review]

File: setup/docker.md [hunk 2]
```diff
@@ -17,7 +17,21 @@ Coder for Docker works with the following platforms:
   [Docker Desktop 20.10](https://www.docker.com/products/docker-desktop).
   **Note**: Coder for Docker requires Windows Subsystem for Linux at this time.
 
-Coder requires x86-64 and does not support ARM-based processors at this time.
+### For systems without x86-64 architecture
```
Comment (khorne3) [inline setup/docker.md:20]: @jawnsy what do you think of the header? Not sure how to phrase this -- I wanted `non-x86-64 architecture systems` but that had too many hyphens :)

**Review by khorne3 — APPROVED** [review]

File: setup/docker.md (hunk 2 above)
Comment (jawnsy) [reply setup/docker.md:20]: I guess "Non-x86_64 systems" might work, avoids the hyphen problem

Trivia: The actual name for this architecture is amd64, and Intel licenses it from AMD, so many systems refer to it with that name. Intel had their own ia64 (Itanium) architecture but it's defunct.

https://en.wikipedia.org/wiki/X86-64

File: setup/docker.md [hunk 3]
```diff
@@ -12,12 +12,27 @@ Coder for Docker works with the following platforms:
 
 - macOS 10.10+ with
   [Docker Desktop 20.10](https://www.docker.com/products/docker-desktop)
+
+  **Note**: If your computer uses an [Apple silicon] processor, you will need to
```
Comment (jawnsy) [inline setup/docker.md:16]: @khorne3 reorganized this to avoid needing a heading, WDYT?

Comment (khorne3) [reply setup/docker.md:16]: @jawnsy looks good to me!

**Review by khorne3 — APPROVED** [review]

**Pull request merged**